src/Nethermind/Nethermind.Core/Caching/LruCache.cs

```diff
@@ -3,170 +3,226 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Runtime.CompilerServices;
 using Nethermind.Core.Extensions;
 
-namespace Nethermind.Core.Caching
+namespace Nethermind.Core.Caching;
+
+public sealed class LruCache<TKey, TValue> : ICache<TKey, TValue> where TKey : notnull
 {
-    /// <remarks>
-    /// The array based solution is preferred to lower the overall memory management overhead. The <see cref="LinkedListNode{T}"/> based approach is very costly.
-    /// </remarks>
-    /// <summary>
-    /// https://stackoverflow.com/questions/754233/is-it-there-any-lru-implementation-of-idictionary
-    /// </summary>
-    public class LruCache<TKey, TValue> : ICache<TKey, TValue> where TKey : notnull
-    {
-        private readonly int _maxCapacity;
-        private readonly Dictionary<TKey, LinkedListNode<LruCacheItem>> _cacheMap;
-        private readonly LinkedList<LruCacheItem> _lruList;
+    private readonly int _maxCapacity;
+    private readonly Dictionary<TKey, LinkedListNode<LruCacheItem>> _cacheMap;
+    private LinkedListNode<LruCacheItem>? _first;
```

Comment:
field is called `first` but logically it actually is `last`?

Reply:
Renamed for clarity